dask/dataframe/shuffle.py

```diff
@@ -611,6 +623,33 @@ def barrier(args):
     return 0
 
 
+def cleanup_partd_files(p, keys):
+    """
+    Cleanup the files in a partd.File dataset.
+
+    Parameters
+    ----------
+    p : partd.Interface
+        File or Encode wrapping a file should be OK.
+    exc_only : bool, default True
+        Whether to cleanup if and only if there's an exception.
+    """
+    import partd
+
+    if isinstance(p, partd.Encode):
+        maybe_file = p.partd
+    else:
+        maybe_file
+
+    if isinstance(maybe_file, partd.File):
+        path = maybe_file.path
+    else:
+        path = None
+
+    if path:
+        shutil.rmtree(path, ignore_errors=True)
```

Comment:
It's low priority, but some folks might create their own PartD objects here.  In the future we might want a more robust solution to finding a File object.